ls1intum/Artemis
# Enhancement/Programming Exercise/Enable Trigger All Button when build was triggered

### Checklist

### Motivation and Context



Fixes issue https://github.com/ls1intum/Artemis/issues/1029.
The trigger all button is re-enabled when a test-cases-changed message is sent to the client.
This means that the button is not enabled if the test-cases dirty flag was not set before.

### Description


Add a new service that can be used to query information about running exercise build runs.
This way the trigger all button will be enabled once `ProgrammingSubmissionService.triggerInstructorBuildForExercise` has triggered all builds.
The instructor will receive a notification about the fact that all builds were triggered.

Notes:
- When the user reloads Artemis after the build was started, we don't know about the build anymore, so the trigger all button will be enabled regardless of the build run state
- The trigger all button will be enabled when all builds were triggered, not when all triggered builds received a result (even though this might be preferable)

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration > programming exercise > participation
3. Click trigger all button, should enter loading state, then when all builds are triggered should become enabled again.

### Screenshots



![reenable-trigger](https://user-images.githubusercontent.com/28230611/68592719-0132d180-0494-11ea-9a81-aef57453505e.gif)
